src/sonic-host-services/scripts/caclmgrd

```diff
@@ -416,16 +432,16 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
                         # try to do it now. We attempt to determine heuristically based on
                         # whether the src or dst IP of this rule is an IPv4 or IPv6 address.
                         if not table_ip_version:
-                            if self.is_rule_ipv6(rule_props):
+                            if self.is_rule_ipv6(rule_props, ethertype):
```

Comment:
It seems redundant to pass `ethertype` to these functions along with `rule_props`, as it is already a part of `rule_props`, and it is not used elsewhere. It might be cleaner (but a tiny bit less efficient) to create a new function like:

```python3
def get_ethertype_from_rule_props(rule_props):
    ethertype_str = rule_props.get('ETHER_TYPE', '0')
    ethertype = int(ethertype_str, 0)
    return ethertype
```

Then you can call this function inside `is_rule_ipv4()` and `is_rule_ipv6()`, and also call it above replacing lines 418-423 as follows:

```python3
                        ethertype = get_ethertype_from_rule_props(rule_props)
                        if ethertype not in [0, 0x0800, 0x86dd]:
                                log_error("rule_props for rule_id {} does not have valid/supported ethertype.".format(rule_id))
                                continue
```

Thoughts?